lib/widgets/inbox.dart

```diff
@@ -421,15 +454,16 @@ class _StreamSection extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final subscription = PerAccountStoreWidget.of(context).subscriptions[data.streamId]!;
-    final header = _StreamHeaderItem(
+    final header = StreamHeaderItem(
       subscription: subscription,
       count: data.count,
       hasMention: data.hasMention,
       collapsed: collapsed,
       pageState: pageState,
     );
+    final stickyHeader = header.copyWith(parentContext: context);
```

Comment:
Yeah, sure. It seems redundant to have two headers, but I couldn't find a better way. If we use one header as follows:

```dart
final header = StreamHeaderItem(
      subscription: subscription,
      count: data.count,
      hasMention: data.hasMention,
      collapsed: collapsed,
      pageState: pageState,
      sectionContext: context,
    );
    return StickyHeaderItem(
      header: header,
      child: Column(children: [
        header,
        if (!collapsed) ...data.items.map((item) {
        
```
Then it will work fine only on sticky headers. Collapsing a section with its header in the middle of the screen will also jump to the top of the screen as both are the same instances of the header.

<img src="https://github.com/zulip/zulip-flutter/assets/59946442/27ca80e6-e994-457d-8e39-82433d2bc777" width="300" />

So to auto-scroll to the top by only by tapping the sticky header, there should be two instances of the header, with the `context` only passed to the sticky one.

Reply:
That's the behavior that should be controlled by the `alignmentPolicy` I mentioned. What do its docs say?

Reply:
Ok thanks, got it!